resources/interactive.mjs

```diff
@@ -0,0 +1,213 @@
+        label.appendChild(document.createTextNode(formatTestName(suite.name)));
+        li.appendChild(label);
+        label.htmlFor = checkbox.id;
+
+        const testList = document.createElement('ol');
+        for (let testIndex = 0; testIndex < suite.tests.length; testIndex++) {
+            const testItem = document.createElement('li');
+            const test = suite.tests[testIndex];
+            const anchor = document.createElement('a');
+            anchor.id = suite.name + '-' + test.name;
+            test.anchor = anchor;
+            anchor.appendChild(document.createTextNode(formatTestName(suite.name, test.name)));
+            testItem.appendChild(anchor);
+            testList.appendChild(testItem);
+        }
+        li.appendChild(testList);
+
+        ol.appendChild(li);
+    }
+
+    control.appendChild(ol);
+
+    let button = document.createElement('button');
+    button.textContent = 'Step';
+    button.onclick = onStep;
+    control.appendChild(button);
+
+    button = document.createElement('button');
+    button.textContent = 'Run';
+    button.id = 'runSuites';
+    button.onclick = onRunSuites;
+    control.appendChild(button);
+
+    button = document.createElement('button');
+    button.textContent = 'Select all';
+    button.onclick = () => {
+        for (var suiteIndex = 0; suiteIndex < suites.length; suiteIndex++) {
+            suites[suiteIndex].disabled = false;
+            checkboxes[suiteIndex].checked = true;
+        }
+    };
+    control.appendChild(button);
+
+    button = document.createElement('button');
+    button.textContent = 'Unselect all';
+    button.onclick = () => {
+        for (var suiteIndex = 0; suiteIndex < suites.length; suiteIndex++) {
+            suites[suiteIndex].disabled = true;
+            checkboxes[suiteIndex].checked = false;
+        }
+
+    };
+    control.appendChild(button);
+
+    return control;
+}
+
+const searchParams = new URLSearchParams(window.location.search);
+
+function disableAllSuitesExcept(suiteName)
+{
+    let foundMatching = false;
+    Suites.forEach(element => {
+        if (element.name !== suiteName)
+            element.disabled = true;
+        else
+            foundMatching = true;
+    });
+    if (!foundMatching)
+        throw Error(`No matching suite for: "${suiteName}"`)
+
+}
+
+function startTest()
+{
+    const queryParam = searchParams.get('suite');
+    if (queryParam !== undefined)
+        disableAllSuitesExcept(queryParam);
+
+    const benchmarkClient = {
+        _stepPromise: undefined,
+        _stepPromiseResolve: undefined,
+        isRunning: false,
+        isStepping: false,
+        willStartFirstIteration()
+        {
+            if (this.isRunning)
+                throw Error('Runner was not stopped before starting;');
+            this.isRunning = true;
+            if (this.isStepping)
+                this._stepPromise = this._newStepPromise();
+        },
+        step()
```

Comment:
It's a bit confusing that this is a part of `benchmarkClient` since it's not a part of the client interface.
Maybe it's better to define this as a lambda, which knows how to update variables of `benchmarkClient`.
WDYT?